### DISCUSSION TITLE
Testing that an action also works in GitHub Enterprise Server/Cloud

### DISCUSSION THREAD

**Discussion by cmarqu** [community/community#47057] [Enterprise]
### Select Topic Area

Question

### Body

Is there a way for a developer of an action developed in GitHub Free to check that it also runs in GitHub Enterprise Server (or Enterprise Cloud)?

Depending on  the specific versions involved, there are several incompatibilities, e.g. https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ (the new syntax is not supported in GHES 3.6), or the difference between `${{ github.action_path }}` and `$GITHUB_ACTION_PATH` (https://github.com/actions/runner/issues/716#issuecomment-795238933).

A normal CI setup will not find such problems. Any recommendations other than "let users of GHES/GHEC do the testing for you"?

The concrete example triggering this question is https://github.com/hynek/build-and-inspect-python-package/issues/37.

Thanks in advance for any advice.

**Comment by killermoehre** [comment]
Piggybacking here: can the GHES version be provided as environment variable or via the `github` context? Right now, you have to query the API of the GHES for this, which is costly, if you want to distinguish in something like a python script over and over again.

**Comment by rossjrw** [comment]
I suspect this will never receive an answer, but noting that I'm also interested in a solution for this.

**Reply by cmarqu** [reply]
FWIW, this answer (thanks to the author!) may help you: https://github.com/hynek/build-and-inspect-python-package/issues/37#issuecomment-1432113112
